### ISSUE TITLE
RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by harryi3t** [felixge/node-combined-stream#37]
When there is a large no of `appends`, I get this `maximum stack size exceeded` exception.

```js
var CombinedStream = require('combined-stream'),
    fs = require('fs'),
    combinedStream = CombinedStream.create();

new Array(1e4).fill(1).forEach(function () {
    combinedStream.append(function(next) {
        next('Content-Disposition: form-data; name="foo" value="bar"');
    });
});

// Throws RangeError: Maximum call stack size exceeded
combinedStream.pipe(fs.createWriteStream('combined.txt')); 
```

This is because [there is a recursion loop](https://github.com/felixge/node-combined-stream/blob/master/lib/combined_stream.js#L68) between `_getNext` and `_pipeNext`

[request](https://www.npmjs.com/package/request) module uses [form-data](https://www.npmjs.com/package/form-data) module, which in turn uses this module. So this is making the `request` module crash.

Note: Originally this issue was observed on [request](https://github.com/request/request/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+call+stack+size) module
But I am not sure where this should be fixed.

Linked Issues:
Request: https://github.com/request/request/issues/1561
Form-Data: https://github.com/form-data/form-data/issues/375

**Comment by apechimp** [comment]
Nice work finding the root cause of that bug! I think it's appropriate to fix the issue here. Can you write up a pull request to fix that?

**Comment by jmendeth** [comment]
This was fixed by #38
